### PR TITLE
fix: rwkv cpp windows failing

### PR DESCRIPTION
**Pull request by hlhr202**

### CODE REVIEW

**Comment by hlhr202** [comment]
@yorkzero831 Hi I v modified for is_first flag by adding an inference session here.

**Comment by yorkzero831** [comment]
@hlhr202 Seems you are working on session management feature, looks good

**Comment by hlhr202** [comment]
But this bug just be quite weird. Since I just made a random copy and translate the python inference code from rwkv.cpp. I did not realize the pointer issue of creating the optional logits and state input for evaluation (cuz rwkv.cpp is creating optional input on python side). probably there may be some memory alignment issue here? I m a noobie rust so still cant figure out the real problem...

**Comment by hlhr202** [comment]
and the linux binary is just working well. so I can only guess the reason that compiler in different platforms generated bytecode in different logic.

**Comment by yorkzero831** [comment]
> But this bug just be quite weird. Since I just made a random copy and translate the python inference code from rwkv.cpp. I did not realize the pointer issue of creating the optional logits and state input for evaluation (cuz rwkv.cpp is creating optional input on python side). probably there may be some memory alignment issue here? I m a noobie rust so still cant figure out the real problem...

Could not agree more, I worked on "native" rust version of rwkv.cpp https://github.com/yorkzero831/rwkv-rs (1/3 performance compare to rwkv.cpp, almost dead), I also faced same problem of migrating python code, I gived up using optional and use vec instead, I thought it should be much more rust way

**Comment by yorkzero831** [comment]
looking forward to next release version lol

**Comment by yorkzero831** [comment]
@hlhr202 btw I sent a private message on twitter to you
